### PR TITLE
pinocchio: deactivate one failing test under aarch64-linux

### PR DESCRIPTION
**Pull request by nim65s**
## Description of changes

Disable one failing test-cpp-algorithm-utils-force on aarch64-linux

ref. https://github.com/NixOS/nixpkgs/pull/315303#issuecomment-2152867279
ref. https://github.com/stack-of-tasks/pinocchio/issues/2277
cc. @wegank 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/libraries/pinocchio/default.nix [hunk 1]
```diff
@@ -24,6 +24,12 @@ stdenv.mkDerivation (finalAttrs: {
     hash = "sha256-h4NzfS27+jWyHbegxF+pgN6JzJdVAoM16J6G/9uNJc4=";
   };
 
+  # test failure, ref https://github.com/stack-of-tasks/pinocchio/issues/2277
+  prePatch = lib.optionals (stdenv.isLinux && stdenv.isAarch64) ''
+    substituteInPlace unittest/add_pinocchio_unit_test(force) \
+      --replace "add_pinocchio_unit_test(force)" ""
```
Comment (kirillrdy) [inline pkgs/development/libraries/pinocchio/default.nix:30]:
```suggestion
      --replace-fail "add_pinocchio_unit_test(force)" ""
```

File: pkgs/development/libraries/pinocchio/default.nix [hunk 2]
```diff
@@ -24,6 +24,12 @@ stdenv.mkDerivation (finalAttrs: {
     hash = "sha256-h4NzfS27+jWyHbegxF+pgN6JzJdVAoM16J6G/9uNJc4=";
   };
 
+  # test failure, ref https://github.com/stack-of-tasks/pinocchio/issues/2277
+  prePatch = lib.optionals (stdenv.isLinux && stdenv.isAarch64) ''
+    substituteInPlace unittest/add_pinocchio_unit_test(force) \
```
Comment (wegank) [inline pkgs/development/libraries/pinocchio/default.nix:29]: The filename doesn't seem right...

Comment (nim65s) [reply pkgs/development/libraries/pinocchio/default.nix]: Hum, I can't even copy-paste at this hour. Thanks !

**Review by wegank — APPROVED** [review]

File: pkgs/development/libraries/pinocchio/default.nix [hunk 3]
```diff
@@ -24,6 +24,12 @@ stdenv.mkDerivation (finalAttrs: {
     hash = "sha256-h4NzfS27+jWyHbegxF+pgN6JzJdVAoM16J6G/9uNJc4=";
   };
 
+  # test failure, ref https://github.com/stack-of-tasks/pinocchio/issues/2277
+  prePatch = lib.optionals (stdenv.isLinux && stdenv.isAarch64) ''
```
Comment (wegank) [inline pkgs/development/libraries/pinocchio/default.nix:28]:
```suggestion
  prePatch = lib.optionalString (stdenv.isLinux && stdenv.isAarch64) ''
```

**Review by wegank — APPROVED** [review]

**Comment by kirillrdy** [comment]
@nim65s second commit doesn't comply commit conventions, can you either update commit message or squash commits ?

**Comment by nim65s** [comment]
Thanks, done

**Pull request merged**